File: monteCarloMethod.py
import random

n = int(input("Enter the number of points to generate: "))
count = 0

for i in range(n):
    x = random.random()
    y = random.random()

    if x**2 + y**2 <= 1:
        count += 1

pi = 4 * count / n

print("Approximation of pi:", pi)

File: monteCarloMethod.c
#include <stdio.h>
#include <stdlib.h>
#include <time.h>
#include <math.h>

int main() {
    int n;
    double x, y;
    int count = 0;
    double pi;

    printf("Enter the number of points to generate: ");
    scanf("%d", &n);

    srand(time(NULL));

    for (int i = 0; i < n; i++) {
        x = (double)rand() / RAND_MAX;
        y = (double)rand() / RAND_MAX;

        if (x * x + y * y <= 1) {
            count++;
        }
    }

    pi = 4.0 * count / n;

    printf("Approximation of pi: %f\n", pi);

    return 0;
}
